### PR TITLE
stats: DogStatsd sink supports custom metric name prefix

### LINKED ISSUE #4390: DogStatsdSink to support Prefixes
**Issue Template**

*Title*: DogStatsdSink to support Prefixes

*Description*:
Currently, StatsD sink supports Prefixes override for users that do not want metrics to start with `Envoy`.

DogStatsd does not. Proto for DogStatsd: https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/metrics/v2/stats.proto#L193

Statsd sink support: https://github.com/envoyproxy/envoy/blob/master/api/envoy/config/metrics/v2/stats.proto#L185

### PR DESCRIPTION
**Pull request by taiki45**
*Description*:

DogStatsd sink supports custom global metric name prefix like [StatsdSink does](https://github.com/envoyproxy/envoy/pull/3029).

Fixes #4390 

*Risk Level*: Low, small optional feature.
*Testing*: Automated tests passed locally. Manually tested with statsd_exporter and Prometheus.
*Docs Changes*: Included.
*Release Notes*: Included.

### CODE REVIEW

**Comment by junr03** [comment]
@mrice32 do you mind taking a first pass?

File: api/envoy/config/metrics/v2/stats.proto [hunk 1]
```diff
@@ -200,6 +200,10 @@ message DogStatsdSink {
   }
 
   reserved 2;
+
+  // Optional custom metric name prefix the same as :ref:`StatsdSink's prefix field
```
Comment (mrice32) [inline api/envoy/config/metrics/v2/stats.proto]: nit: maybe rephrase to something like:

```
// Optional custom metric name prefix. See 
// :ref:`StatsdSink's prefix field <envoy_api_field_config.metrics.v2.StatsdSink.prefix>`
// for more details.

**Comment by taiki45** [comment]
Right, I agree we should have tests in that area. I'll try adding them in this PR.

**Comment by taiki45** [comment]
Sorry to be late. I added some tests for custom metric prefix.

**Comment by mattklein123** [comment]
@taiki45 can you merge master to bring in the bazel build cache fix and also fix merge conflicts? I can look after that.

File: api/envoy/config/metrics/v2/stats.proto (hunk 1 above)
Comment (mrice32) [reply api/envoy/config/metrics/v2/stats.proto]: Reminder to fix this nit :)

Comment (taiki45) [reply api/envoy/config/metrics/v2/stats.proto]: Oh, I though I fixed it... I'll fix it now.

Comment (taiki45) [reply api/envoy/config/metrics/v2/stats.proto]: Thanks, now I fixed this.
